### PR TITLE
Chapter 3: Mostly Static Pages

### PR DESCRIPTION
**Pull request by megatimranikram**

### CODE REVIEW

**Comment by megatimranikram** [comment]
Ready

File: Gemfile [hunk 1]
```diff
@@ -3,15 +3,16 @@ git_source(:github) { |repo| "https://github.com/#{repo}.git" }
 
 ruby "2.7.1"
 
+gem "dotenv-rails", groups: [:development, :test]
```
Comment (manhnt-0960) [inline Gemfile]: please move this gem down to group :development, :test

File: Gemfile [hunk 2]
```diff
@@ -29,9 +30,8 @@ group :test do
...
 end
 
+gem "tzinfo-data", platforms: [:mingw, :mswin, :x64_mingw, :jruby]
 group :development, :test do
   gem "rubocop", "~> 0.74.0", require: false
   gem "rubocop-rails", "~> 2.3.2", require: false
-end
-
-gem "tzinfo-data", platforms: [:mingw, :mswin, :x64_mingw, :jruby]
+end
```
Comment (manhnt-0960) [inline Gemfile]: end line

File: app/helpers/application_helper.rb [hunk 3]
```diff
@@ -1,2 +1,4 @@
 module ApplicationHelper
-end
+    def full_title page_title = ""
```
Comment (manhnt-0960) [inline app/helpers/application_helper.rb]: please check indent again

File: app/helpers/application_helper.rb [hunk 4]
```diff
@@ -1,2 +1,4 @@
...
+        base_title = "Ruby on Rails Tutorial Sample App"
+        page_title.blank? ? base_title : "#{page_title} | #{base_title}"
```
Comment (manhnt-0960) [inline app/helpers/application_helper.rb]: you missing `end` keyword

File: app/views/static_pages/home.html.erb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+<% provide :title, "Home" %>
+
+<h1>Sample App</h1>
+<p>
+  This is the home page for the
+  <%= link_to "Ruby on Rails Tutorial", Settings.links.rails_tutorial.home_page %>.
+</p>
```
Comment (manhnt-0960) [inline app/views/static_pages/home.html.erb]: add new line at end of file, please check with another files

**Comment by megatimranikram** [comment]
Ready sir

File: Gemfile (hunk 1 above)
Comment (megatimranikram) [reply Gemfile]: done sir

File: Gemfile (hunk 2 above)
Comment (megatimranikram) [reply Gemfile]: done sir

File: app/helpers/application_helper.rb (hunk 3 above)
Comment (megatimranikram) [reply app/helpers/application_helper.rb]: done sir

File: app/helpers/application_helper.rb (hunk 4 above)
Comment (megatimranikram) [reply app/helpers/application_helper.rb]: done sir

File: app/views/static_pages/home.html.erb (hunk 5 above)
Comment (megatimranikram) [reply app/views/static_pages/home.html.erb]: done sir

**Comment by megatimranikram** [comment]
ready

File: app/helpers/application_helper.rb [hunk 6]
```diff
@@ -1,2 +1,6 @@
...
+    end
+
```
Comment (ngocvt-0484) [inline app/helpers/application_helper.rb]: remove blank line

File: app/helpers/application_helper.rb (hunk 3 above)
Comment (ngocvt-0484) [reply app/helpers/application_helper.rb]: not fix yet

File: config/routes.rb [hunk 7]
```diff
@@ -1,3 +1,5 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
-end
+  root "static_pages#home"
+
+  get "static_pages/help"
+end
```
Comment (ngocvt-0484) [inline config/routes.rb]: please add endline No12
https://docs.google.com/document/d/104Csp4-vamVos5DEbi372nLBEfmqhj7h1ENYO8ebjCo/edit

File: app/helpers/application_helper.rb (hunk 6 above)
Comment (megatimranikram) [reply app/helpers/application_helper.rb]: done ms

File: config/routes.rb (hunk 7 above)
Comment (megatimranikram) [reply config/routes.rb]: done ms

**Comment by megatimranikram** [comment]
ready

File: .gitignore [hunk 8]
```diff
@@ -32,13 +32,18 @@
...
 /yarn-error.log
 yarn-debug.log*
 .yarn-integrity
-/config/credentials.yml.enc
-/config/database.yml
-/test
+
+config/settings.local.yml
+config/settings/*.local.yml
+config/environments/*.local.yml
```
Comment (saobv-2223) [inline .gitignore]: add an empty line below lin 49

File: app/controllers/static_pages_controller.rb [hunk 9]
```diff
@@ -0,0 +1,5 @@
+class StaticPagesController < ApplicationController
+  def home; end
+
+  def help; end
+end
```
Comment (saobv-2223) [inline app/controllers/static_pages_controller.rb]: add an empty line below line 5

File: app/helpers/application_helper.rb [hunk 10]
```diff
@@ -1,2 +1,5 @@
 module ApplicationHelper
-end
+    def full_title page_title = ""
+        base_title = "Ruby on Rails Tutorial Sample App"
+        page_title.blank? ? base_title : "#{page_title} | #{base_title}"
+    end
```
Comment (saobv-2223) [inline app/helpers/application_helper.rb]:
```suggestion
  def full_title page_title
    base_title = "Ruby on Rails Tutorial Sample App"
    page_title.blank? ? base_title : "#{page_title} | #{base_title}"
  end
end
```

File: app/javascript/packs/application.js [hunk 11]
```diff
@@ -1,17 +1,4 @@
...
-require('channels')
-
-
-// Uncomment to copy all static images under ../images to the output folder and reference
-// them with the image_pack_tag helper in views (e.g <%= image_pack_tag 'rails.png' %>)
-// or the `imagePath` JavaScript helper below.
-//
-// const images = require.context('../images', true)
-// const imagePath = (name) => images(name, true)
+require('channels')
```
Comment (saobv-2223) [inline app/javascript/packs/application.js]: add an empty line below line 4

File: config/application.rb [hunk 12]
```diff
@@ -2,18 +2,12 @@
...
-
-    # Settings in config/environments/* take precedence over those specified here.
-    # Application configuration can go into files in config/initializers
-    # -- all .rb files in that directory are automatically loaded after loading
-    # the framework and any gems in your application.
+    config.i18n.available_locales = [:en, :ru]
+    config.i18n.default_locale = :en
   end
-end
+end
```
Comment (saobv-2223) [inline config/application.rb]: add an empty line below line 13

File: config/initializers/config.rb [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+  
+    # Validate presence and type of specific config values. Check https://github.com/dry-rb/dry-validation for details.
+    #
+    # config.schema do
+    #   required(:name).filled
+    #   required(:age).maybe(:int?)
+    #   required(:email).filled(format?: EMAIL_REGEX)
+    # end
+  
+  end
```
Comment (saobv-2223) [inline config/initializers/config.rb]: add an empty line below line 55

File: config/locales/en.yml [hunk 14]
```diff
@@ -1,33 +1,2 @@
...
-#
-# en:
-#   'true': 'foo'
-#
-# To learn more, please read the Rails Internationalization guide
-# available at https://guides.rubyonrails.org/i18n.html.
-
 en:
-  hello: "Hello world"
+  hello: "Hello world"
```
Comment (saobv-2223) [inline config/locales/en.yml]: add an empty line below line 2

File: config/settings.yml [hunk 15]
```diff
@@ -0,0 +1,5 @@
+links:
+  rails_tutorial:
+    home_page: https://www.railstutorial.org/
+    help_page: http://www.railstutorial.org/help
+    book_page: https://www.railstutorial.org/book
```
Comment (saobv-2223) [inline config/settings.yml]: add an empty line below line 5

File: .gitignore (hunk 8 above)
Comment (megatimranikram) [reply .gitignore]: done sir

File: app/controllers/static_pages_controller.rb (hunk 9 above)
Comment (megatimranikram) [reply app/controllers/static_pages_controller.rb]: done sir

File: app/helpers/application_helper.rb (hunk 10 above)
Comment (megatimranikram) [reply app/helpers/application_helper.rb]: done sir

File: app/javascript/packs/application.js (hunk 11 above)
Comment (megatimranikram) [reply app/javascript/packs/application.js]: done sir

File: config/initializers/config.rb (hunk 13 above)
Comment (megatimranikram) [reply config/initializers/config.rb]: done sir

File: config/locales/en.yml (hunk 14 above)
Comment (megatimranikram) [reply config/locales/en.yml]: done sir

File: config/settings.yml (hunk 15 above)
Comment (megatimranikram) [reply config/settings.yml]: done sir

**Comment by megatimranikram** [comment]
ready

File: app/helpers/application_helper.rb [hunk 16]
```diff
@@ -1,2 +1,6 @@
 module ApplicationHelper
-end
+  def full_title page_title
+    base_title = "Ruby on Rails Tutorial Sample App"
+    page_title.blank? ? base_title : "#{page_title} | #{base_title}"
+  end
+end
```
Comment (chungpv-1008) [inline app/helpers/application_helper.rb]: add endline for this file

File: config/locales/en.yml [hunk 17]
```diff
@@ -1,33 +1,3 @@
...
-#
-# en:
-#   'true': 'foo'
-#
-# To learn more, please read the Rails Internationalization guide
-# available at https://guides.rubyonrails.org/i18n.html.
-
 en:
   hello: "Hello world"
+
```
Comment (chungpv-1008) [inline config/locales/en.yml]: add endline for this file

File: config/settings.yml [hunk 18]
```diff
@@ -0,0 +1,6 @@
...
+
```
Comment (chungpv-1008) [inline config/settings.yml]: add endline for this file

File: app/helpers/application_helper.rb (hunk 16 above)
Comment (megatimranikram) [reply app/helpers/application_helper.rb]: done sir

File: config/locales/en.yml (hunk 17 above)
Comment (megatimranikram) [reply config/locales/en.yml]: done sir

File: config/settings.yml (hunk 18 above)
Comment (megatimranikram) [reply config/settings.yml]: done sir

**Comment by megatimranikram** [comment]
ready

File: config/locales/en.yml (hunk 17 above)
Comment (saobv-2223) [inline config/locales/en.yml]: add an empty line below line 3

File: config/initializers/config.rb [hunk 19]
```diff
@@ -0,0 +1,56 @@
...
+
```
Comment (saobv-2223) [inline config/initializers/config.rb]: add an empty line below line 56

File: config/locales/en.yml (hunk 17 above)
Comment (megatimranikram) [reply config/locales/en.yml]: done sir

File: config/initializers/config.rb (hunk 19 above)
Comment (megatimranikram) [reply config/initializers/config.rb]: done sir

**Comment by megatimranikram** [comment]
ready

File: config/initializers/config.rb (hunk 19 above)
Comment (saobv-2223) [reply config/initializers/config.rb]: image

File: config/locales/en.yml (hunk 17 above)
Comment (saobv-2223) [reply config/locales/en.yml]: image

Comment (megatimranikram) [reply config/locales/en.yml]: done sir

**Comment by megatimranikram** [comment]
ready

**Review by saobv-2223 — COMMENTED** [review]
merge

**Pull request merged**